rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 2*time.Minute, "Maximum time a query may take before being aborted.")
+)
+
+var queryTimeoutErr = fmt.Errorf("query timeout after %v", *queryTimeout)
```

Comment:
golint wants this as `errQueryTimeout`.

In different news, as we want a variable timeout later, it might not be the best to have a pregenerated string here. When we introduce configurable per-query timeouts, we might forget that the error contains the default timeout. Also, the actual time after which we time out might be slightly higher than the flag value. So I guess I'd go for dynamic creation of the error string.